src/view/dataElements/xdmObject/helpers/array/validate.js

```diff
@@ -34,11 +34,21 @@ export default ({
 
   if (items) {
     const itemErrors = items.map(itemFormStateNode => {
-      return validate({
+      let itemIsPopulated = false;
+      let itemError = validate({
         formStateNode: itemFormStateNode,
-        isParentAnArray: true,
-        notifyParentOfDataPopulation: confirmDataPopulatedAtCurrentOrDescendantNode
+        confirmDataPopulatedAtCurrentOrDescendantNode() {
+          itemIsPopulated = true;
+          confirmDataPopulatedAtCurrentOrDescendantNode();
+        }
       });
+      if (!itemIsPopulated && !itemError) {
+        itemError = {
+          value:
+            "Items within arrays must not be empty. Please populate or remove the item."
+        };
+      }
+      return itemError;
```

Comment:
This validation of empty items within arrays was moved here from xdmObject/helpers/validate.js. It really belonged here because it's an array-specific validation.